kubernetes-full.yml

```diff
@@ -0,0 +1,293 @@
+    matchLabels:
+      app: api
+  template:
+    metadata:
+      labels:
+        app: api
+      name: api
+    spec:
+      containers:
+      - env:
+        - name: MONGODB_DATABASE
+          value: sorry-cypress
+        - name: MONGODB_URI
+          value: mongodb://mongo-service:27017
+        image: agoldis/sorry-cypress-api:cypress-v5
+        imagePullPolicy: "Always"
+        name: api
+        ports:
+        - containerPort: 4000
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 4000
+          periodSeconds: 10
+          timeoutSeconds: 5
+          successThreshold: 2
+          failureThreshold: 5
+        resources:
+          requests:
+            memory: "128M"
+            cpu: 0.2
+          limits:
+            memory: "512M"
+            cpu: 0.5
+      restartPolicy: Always
+      serviceAccountName: ""
+      volumes: null
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: director
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: director
+  template:
+    metadata:
+      labels:
+        app: director
+      name: director
+    spec:
+      containers:
+      - env:
+        - name: DASHBOARD_URL
+          value: [YOUR-DASHBOARD-URL-HERE]
+        - name: EXECUTION_DRIVER
+          value: ../execution/mongo/driver
+        - name: MONGODB_DATABASE
+          value: sorry-cypress
+        - name: MONGODB_URI
+          value: mongodb://mongo-service:27017
+        - name: SCREENSHOTS_DRIVER
+          value: ../screenshots/s3.driver
+        - name: S3_BUCKET
+          value: sendible-cypress
+        - name: S3_REGION
+          value: us-east-1
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+              secretKeyRef:
+                name: cypress-s3-secrets
+                key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+              secretKeyRef:
+                name: cypress-s3-secrets
+                key: AWS_SECRET_ACCESS_KEY
+        image: agoldis/sorry-cypress-director:cypress-v5
+        imagePullPolicy: "Always"
+        name: director
+        ports:
+        - containerPort: 1234
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 1234
+          periodSeconds: 10
+          timeoutSeconds: 5
+          successThreshold: 2
+          failureThreshold: 5
+        resources:
+          requests:
+            memory: "128M"
+            cpu: 0.2
+          limits:
+            memory: "512M"
+            cpu: 0.5
+      restartPolicy: Always
+      serviceAccountName: ""
+      volumes: null
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dashboard
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dashboard
+  template:
+    metadata:
+      labels:
+        app: dashboard
+      name: dashboard
+    spec:
+      containers:
+      - env:
+        - name: GRAPHQL_SCHEMA_URL
+          value: [YOUR-API-URL-HERE]
```

Comment:
Is it possible to use some kind of "internal DNS" to route traffic to API service? I have 0 kubernetes experience

Reply:
Yes, that's possible. I actually already do that on line 130. The director pod accesses mongo internally.
If that's what's supposed to happen here then I'll adjust.

Reply:
Yes, `dashboard` pull graphql schema definition from the API service

Reply:
So it looks like the dashboard also relies on this URL when the end user visits the dashboad through a browser.
Changing this to the internally-accessible dns entry prevents the end user from loading their data.

So I believe the right thing to do in this instance is indeed to call the external API URL.

Reply:
Oh right, it need to access the API from the user-space